src/main/java/training/busboard/web/Website.java

```diff
@@ -1,28 +1,47 @@
 package training.busboard.web;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.servlet.ModelAndView;
+import training.busboard.ApiServices.*;
+import training.busboard.Models.Bus;
+import training.busboard.Models.PostcodeError;
+
+import java.util.List;
 
 @Controller
 @EnableAutoConfiguration
 public class Website {
 
+    Logger LOGGER = LogManager.getLogger("ControllerLogging");
+
     @RequestMapping("/")
     ModelAndView home() {
         return new ModelAndView("index");
     }
 
     @RequestMapping("/busInfo")
     ModelAndView busInfo(@RequestParam("postcode") String postcode) {
-        return new ModelAndView("info", "busInfo", new BusInfo(postcode)) ;
+        ApiRequestManagerService apiService = new ApiRequestManagerService();
+        List<Bus> buses = apiService.getBusStopInfo(postcode);

```

Comment:
Are we doing any validation of the postcode entered by the user? It's best not to trust input from a user in case they (accidentally or maliciously) send something badly formed which doesn't do what you intended. To be honest, I can't think of anything really damaging you could do with this field, but maybe a creative hacker could. 🙂 

You could check that it only contains letters and spaces. You could also URL-encode it before using it in the postcode API request.